### PR TITLE
Refactor/add contract wait

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
Refactoring `addContractWait`:

Removed usage of `getBlock`;
Removed `confirmations` param from `contract.wait`;
Added `timeout` param to `contract.wait`;
Passing `transactionId` to `getTransaction`

**Related issue(s)**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/contracts/src.ts/index.ts [hunk 1]
```diff
@@ -354,10 +352,10 @@ function buildEstimate(contract: Contract, fragment: FunctionFragment): Contract
     };
 }
 
-function addContractWait(contract: Contract, tx: TransactionResponse) {
+function addContractWait(contract: Contract, tx: ContractTransaction) {
```
Comment (Daniel-K-Ivanov) [inline packages/contracts/src.ts/index.ts:355]: Is there a reason for changing this to `ContractTransaction` from `TransactionResponse`?

File: packages/tests/src.ts/test-contract.spec.ts [hunk 2]
```diff
@@ -340,6 +340,66 @@ describe("Test Contract Transaction Population", function() {
         });
         assert.strictEqual(BigNumber.from(balance).toNumber(), 10000, 'balance mismatch');
     }).timeout(60000);
+
+   it('should have a .wait function', async function() {
```
Comment (Daniel-K-Ivanov) [inline packages/tests/src.ts/test-contract.spec.ts:344]: We should assert the exact log being emitted not only that it is populated (!= null)

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

File: packages/contracts/src.ts/index.ts (hunk 1 above)
Comment (Ivo-Yankov) [reply packages/contracts/src.ts/index.ts]: Actually no, it was left from when I was trying to fix an issue that caused errors if there was no `// ts-ignore`.

File: packages/tests/src.ts/test-contract.spec.ts (hunk 2 above)
Comment (Ivo-Yankov) [reply packages/tests/src.ts/test-contract.spec.ts:344]: I made the test more explicit

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**